frontend/src/v5/store/federations/federations.sagas.ts

```diff
@@ -97,6 +105,64 @@ export function* fetchFederationStats({ teamspace, projectId, federationId }: Fe
 	}
 }
 
+export function* fetchFederationViews({
+	teamspace,
+	projectId,
+	federationId,
+}: FetchFederationViewsAction) {
+	try {
+		const { views }: FetchFederationViewsResponse = yield API.Federations.fetchFederationViews({
+			teamspace,
+			projectId,
+			federationId,
+		});
+		yield put(FederationsActions.fetchFederationViewsSuccess(projectId, federationId, views));
+	} catch (error) {
+		yield put(DialogsActions.open('alert', {
+			currentActions: 'trying to fetch federation views',
+			error,
+		}));
+	}
+}
+
+export function* fetchFederationSettings({
+	teamspace,
+	projectId,
+	federationId,
+}: FetchFederationSettingsAction) {
+	try {
+		const rawSettings = yield API.Federations.fetchFederationSettings({
+			teamspace,
+			projectId,
+			federationId,
+		});
+		const settings = prepareFederationSettingsForFrontend(rawSettings);
+		yield put(FederationsActions.fetchFederationSettingsSuccess(projectId, federationId, settings));
+	} catch (error) {
+		yield put(DialogsActions.open('alert', {
+			currentActions: 'trying to fetch federation settings',
+			error,
+		}));
+	}
+}
+
+export function* updateFederationSettings({
+	teamspace, projectId, federationId, settings, extraSettings,
+}: UpdateFederationSettingsAction) {
+	try {
+		const rawSettings = prepareFederationSettingsForBackend(settings, extraSettings);
```

Comment:
Why are we separating and having extra settings here instead of the whole settings thing?

Reply:
I have done this because some properties (namely `name`, `code`, and `desc`) already exist in the `IFederation` and I thought adding them again as part of the `settings` property could have been redundant. Shall I include these properties in the `IFederationSettings` interface instead, and just pass them along with the rest of the settings?